builder/src/build.ts

```diff
@@ -233,23 +225,47 @@ export namespace Build {
       });
     });
 
-    // Template the CSS index file.
     cssImports = cssImports.sort((a, b) => a.localeCompare(b));
-    let cssContents = '/* This is a generated file of CSS imports */';
-    cssContents +=
-      '\n/* It was generated by @jupyterlab/builder in Build.ensureAssets() */';
-    cssContents += `\n@import url('~${appCSS}');`;
+
+    const styleContents = ['/* This is a generated file of CSS imports */'];
+    styleContents.push(
+      '/* It was generated by @jupyterlab/builder in Build.ensureAssets() */'
+    );
+
+    /**
+     * Converts a css file to the corresponding js import if it exists
+     */
+    function cssJS(cssImport: string): string | undefined {
+      const ext = path.extname(cssImport);
+      if (ext === '.css') {
+        const jsImport = `${cssImport.slice(0, cssImport.length - ext.length)}`;
+        try {
+          require.resolve(jsImport);
+          return jsImport;
+        } catch (err) {
+          /* no-op */
+        }
+      }
+    }
+
+    let jsImport;
     cssImports.forEach(cssImport => {
-      cssContents += `\n@import url('~${cssImport}');`;
+      if ((jsImport = cssJS(cssImport))) {
+        styleContents.push(`import '${jsImport}';`);
+      } else {
+        styleContents.push(`// import '${cssImport}';`);
```

Comment:
```suggestion
        styleContents.push(`import '${cssImport}';`);
```
Import css when we need to, just like in ensure-repo